third_party/amd/lib/TritonAMDGPUTransforms/CanonicalizePointers.cpp

```diff
@@ -442,16 +268,118 @@ PointerCanonicalizer::decomposeOffsetFromExpr(Location loc, Value expr,
   return offsets;
 }
 
-Value PointerCanonicalizer::createTensorPointer(FatPtr fatPtr, Location loc) {
-  Value basePtr = fatPtr.basePtr;
-  Value offset = fatPtr.offset;
+static const std::string kPtrCanonPrefix = "__amdpointercanonicalize.";
+static const std::string kSCFThenRewrittenAttr =
+    kPtrCanonPrefix + "scf-then-rewritten__";
+static const std::string kSCFElseRewrittenAttr =
+    kPtrCanonPrefix + "scf-else-rewritten__";
+static const std::string kSCFIfOpYieldFatPtrOffsets =
+    kPtrCanonPrefix + "scf-if-yield-fatptr-offsets__";
+
+struct FatPointers {
```

Comment:
Can you add some comments regarding what this struct is for and how it is used throughout given it's a very important piece of data structure.